### PR TITLE
API: Simplify by removing the need for functor, use definition instead

### LINKED ISSUE #3: Suggestions: simplify API by passing module in graphql hooks
The current API is very close to the one in `reason-apollo-hooks`, which looks like this:
```reason
module HelloQueryConfig = [%graphql ...];
module HelloQuery = ReasonApolloHooks.Query.Make(HelloQueryConfig);
...
let (query, _) = HelloQuery.use();
```

**UPDATE:**

> One cool feature has recently landed in graphql_ppx_re - definition with parse, query and composeVariables accessible as a tuple on MyGraphqlQuery.definition, see this PR. Maybe it is worth moving there instead of using functors/modules?

> Then it will be possible to have useQuery(MyGraphqlQuery.definition, ~variables=..., ()). And reason-urql are going in that direction, see this PR. I could look into this if you think it will be a good idea 😄

**Proposed change:**
```reason
let result = Graphql.useQuery(~variables=..., MyGraphqlQuery.definition, ());
```

------

~~It seems redundant that you need to create `HelloQuery` module for each of your queries and mutations.~~

~~One simplification could be to have `graphql` hooks as simple functions (as opposed to being created from modules) that accept `HelloQueryConfig` module and extract `parse` and `t` from it. Then it could look like this:~~
```reason
[redacted]
```

~~This is possible due to first-class modules in `OCaml`. This [PR](https://github.com/Astrocoders/reason-apollo-hooks/pull/59) in `reason-apollo-hooks` introduces this change.~~

This is just a suggestion, would love to hear your opinion about it 😄

### PR DESCRIPTION
**Pull request by lessp**
This PR uses the `definition` returned from `graphql_ppx`, which allows us to avoid the extra functor-call, going from:

```reason
module HelloQueryConfig = [%graphql
  {|
    query Hello {
      hello {
        name
      }
    }
  |}
];

module HelloQuery = Graphql.Query.Make(HelloQueryConfig);

/* component */
let%hook status = HelloQuery.use();
```

to:

```reason
module HelloQueryConfig = [%graphql
  {|
    query Hello {
      hello {
        name
      }
    }
  |}
];

/* component */
let%hook status = Graphql.useQuery(HelloQueryConfig.definition, ());
```

Would close #3

### CODE REVIEW

File: src/Query.re [hunk 1]
```diff
@@ -3,68 +3,74 @@ open Revery.UI.React;
...
+  let initialState: status('responseType) = Idle;
+
+  let useQuery = (~variables: option(Yojson.Basic.t)=?, definition, ()) => {
+    let (parse, graphqlQuery, composeVariables) = definition;
 
-  let use = (~variables: option(Yojson.Basic.t)=?, ()) => {
     let%hook (state, dispatch) = Hooks.reducer(~initialState, reducer);
 
-    let query = () => {
+    let queryAndVariables = () => {
```
Comment (despairblue) [inline src/Query.re:62]: I made this a pure function outside of the module btw. Seemed cleaner as it won't shadow `variables` and `query` and shortens the body of `useQuery`:

```
let createBody = (~variables=`Assoc([]), ~query) => {
  let payload =
    `Assoc([("query", `String(query)), ("variables", variables)])
    |> Yojson.Basic.to_string;

  (payload, variables);
};
```

Given `~variables` a default argument will mean it needs to be called like this further down:

```
       let (payload, variables) = createBody(~query, ~variables?);
```

@lessp What do you think?

Comment (lessp) [reply src/Query.re:62]: Agree! Do you mind adding it yourself?

**Comment by lessp** [comment]
@all-contributors please add @lessp for ideas, code, docs

**Comment by lessp** [comment]
@all-contributors please add @despairblue for ideas, code

File: src/Query.re (hunk 1 above)
Comment (lessp) [reply src/Query.re]: Resolved in: fd8d9b9, thanks!

**Pull request merged**